arch/src/lib.rs

```diff
@@ -76,6 +76,7 @@ impl FromStr for CpuProfile {
     fn from_str(s: &str) -> result::Result<Self, Self::Err> {
         match s {
             "host" => Ok(Self::Host),
+            #[cfg(target_arch = "x86_64")]
```

Comment:
I also think that this should be split into `arch/{mod,x86_64, aarch64, riscv64}.rs`

Reply:
Hmm, for some reason this doesn't contain my local state where I did this in a much more generic way. Will try to push that now.